src/seismometer/seismogram.py

```diff
@@ -73,13 +80,36 @@ def __init__(self, config_path: Optional[str | Path] = None, output_path: Option
         self.cohort_cols: list[str] = []
         self.config_path = config_path
 
-        self.load_config(config_path)
+        self.load_config(config_path, definitions=definitions)
 
         self.config.set_output(output_path)
         self.config.output_dir.mkdir(parents=True, exist_ok=True)
         self.dataloader = loader_factory(self.config)
 
-    def load_data(self, predictions=None, events=None):
+    def load_data(
+        self, *, predictions: Optional[pd.DataFrame] = None, events: Optional[pd.DataFrame] = None, reset: bool = False
+    ):
+        """
+        Loads the seismogram data.
+
+        Uses the passed in frames if they are specified, otherwise uses configuration to load data.
+        If data is already loaded, does not change state unless reset is true.
+
+        Parameters
+        ----------
+        predictions : pd.DataFrame, optional
+            The fully prepared predictions dataframe, by default None.
+            Uses this when specified, otherwise loads based on configuration.
+        events : pd.DataFrame, optional
+            The pre-loaded events dataframe, by default None.
+            Uses this when specified, otherwise loads based on configuration.
+        reset : bool, optional
+            Flag when set to true will overwrite existing dataframe, by default False
+        """
+        if self.dataframe and not reset:
+            logger.debug("Data already loaded; pass reset=True to clear data and re-evaluate.")
+            return
+
         self._load_metadata()
 
         self.dataframe = self.dataloader.load_data(predictions, events)
```

Comment:
Do we need any special handling when events is none but predictions is not none?

Reply:
I think the pipeline._add_events code actually handles this sufficiently.

My guess is you hit the same thing I did when trying to test and see what you meant, which is that an events.parquet exists and gets loaded when we specify None (or omit the kwarg).  
So for testing, delete that file and verify the case is supported as you'd want.  

Additionally, let's write up something to more robustly handle the case when predictions has the *_Value and/or *_Time columns that merging wants to add. 
My instinct is to prefer prediction data, and just warn about skipping the merge.

Reply:
Can you write the issue up, then I'll resolve this.

Reply:
Wrote up the issue in #39